### PR TITLE
[Merged by Bors] - 0.10 Pipelined rendering

### PR DESCRIPTION
**Pull request by hymm**
also included a section for Plugin::setup that was added for pipelined rendering.

### CODE REVIEW

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 1]
```diff
@@ -501,6 +501,36 @@ As this brings Bevy closer to full support of Android, there isn't a need anymor
...
+<div class="release-feature-authors">authors: @hymm, @james7132</div>
+
+![Trace with Pipelined Rendering](pipelined-rendering-trace.png)
+
+On multithreaded platforms bevy will now run significantly faster by running simulation and
+rendering in parallel. The renderer was rearchiteched in [bevy 0.6](https://bevyengine.org/news/bevy-0-6/#pipelined-rendering-extract-prepare-queue-render)
+to enable this, but the final step of actually running them in parallel was not done until
+now. There was a bit of tricky work to figure out around enabling running specific rendering
+systems on the main app thread instead of the rendering thread and making sure !Send
+resources aren't accessed from a different thread than they are initialized on.
```
Comment (Trouv) [inline content/news/2023-03-04-bevy-0.10/index.md:515]: I found this sentence kinda hard to parse. Maybe something like this?

```suggestion
now. There were some complications when moving specific rendering
systems from the rendering thread to the main app thread, and making sure !Send
resources aren't accessed from a different thread than they are initialized on.
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 2]
```diff
@@ -501,6 +501,36 @@ As this brings Bevy closer to full support of Android, there isn't a need anymor
...
+
+![Histogram of Many Foxes Frame Time](pipelined-rendering-histogram.png)
+
+In testing different bevy examples, the gains were typically in the 10% to 30% range. As seen in the above histogram the many foxes stress test gains 1.8ms.
```
Comment (Trouv) [inline content/news/2023-03-04-bevy-0.10/index.md:519]: I think a dependent/independent clause comma is needed here more than in other places
```suggestion
In testing different bevy examples, the gains were typically in the 10% to 30% range. As seen in the above histogram, the "many foxes" stress test gains 1.8ms.
```

Comment (Trouv) [inline content/news/2023-03-04-bevy-0.10/index.md:519]: Also - can you be more specific about "gains `1.8ms`"? Do you mean "the stress test speeds up by an average of 1.8ms per frame"? I couldn't really tell from the image

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 3]
```diff
@@ -501,6 +501,36 @@ As this brings Bevy closer to full support of Android, there isn't a need anymor
...
+
+To use pipelined rendering, you just need to add the `PipelinedRenderingPlugin`. If you're
+using `DefaultPlugins` then it will automatically be added for you on all platforms except
+wasm. Bevy does not currently support multithreading on wasm which is needed for this
+feature to work. If you are not using `DefaultPlugins` you can add the plugin manually.
+
+## Added a post-build method on `Plugin`
+
+A optional `setup` method was added to `Plugin` that runs after all the build methods have
```
Comment (Trouv) [inline content/news/2023-03-04-bevy-0.10/index.md:528]:
```suggestion
An optional `setup` method was added to the `Plugin` trait that runs after all the build methods have
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 4]
```diff
@@ -501,6 +501,36 @@ As this brings Bevy closer to full support of Android, there isn't a need anymor
...
+been called. This was required to enable pipelined rendering. It needed to remove the sub
```
Comment (Trouv) [inline content/news/2023-03-04-bevy-0.10/index.md:529]: This "it" could have been referring to either the pipelined rendering or the setup method
```suggestion
been called. This was required to enable pipelined rendering, which needed to remove the sub
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 5]
```diff
@@ -501,6 +501,36 @@ As this brings Bevy closer to full support of Android, there isn't a need anymor
...
+app from the app to send it between the main thread and the rendering thread. But this is
```
Comment (Trouv) [inline content/news/2023-03-04-bevy-0.10/index.md:530]: the but isn't incorrect, but I don't think it's necessary
```suggestion
app from the app to send it between the main thread and the rendering thread. This is
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 6]
```diff
@@ -501,6 +501,36 @@ As this brings Bevy closer to full support of Android, there isn't a need anymor
 
 ![iOS emulator running Bevy](ios%20emulator.png)
 
+## Enable Parallel Pipelined Rendering
+
+<div class="release-feature-authors">authors: @hymm, @james7132</div>
+
+![Trace with Pipelined Rendering](pipelined-rendering-trace.png)
+
+On multithreaded platforms bevy will now run significantly faster by running simulation and
```
Comment (Trouv) [inline content/news/2023-03-04-bevy-0.10/index.md:510]: dep, ind clauses
```suggestion
On multithreaded platforms, bevy will now run significantly faster by running simulation and
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 7]
```diff
@@ -501,6 +501,36 @@ As this brings Bevy closer to full support of Android, there isn't a need anymor
...
+rendering in parallel. The renderer was rearchiteched in [bevy 0.6](https://bevyengine.org/news/bevy-0-6/#pipelined-rendering-extract-prepare-queue-render)
```
Comment (LiamGallagher737) [inline content/news/2023-03-04-bevy-0.10/index.md:511]:
```suggestion
rendering in parallel. The renderer was rearchitected in [bevy 0.6](https://bevyengine.org/news/bevy-0-6/#pipelined-rendering-extract-prepare-queue-render)
```

**Review by Trouv — COMMENTED** [review]
Thanks for the interesting read, I have some minor nitpicks

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 8]
```diff
@@ -501,6 +501,36 @@ As this brings Bevy closer to full support of Android, there isn't a need anymor
...
+to enable this, but the final step of actually running them in parallel was not done until
+now. There was a bit of tricky work to figure out around enabling running specific rendering
+systems on the main app thread instead of the rendering thread and making sure !Send
```
Comment (james7132) [inline content/news/2023-03-04-bevy-0.10/index.md:514]:
```suggestion
systems on the main app thread instead of the rendering thread and making sure `!Send`
```

File: content/news/2023-03-04-bevy-0.10/index.md (hunk 4 above)
Comment (james7132) [reply content/news/2023-03-04-bevy-0.10/index.md:529]: s/which/which is

**Review by james7132 — CHANGES_REQUESTED** [review]

File: content/news/2023-03-04-bevy-0.10/index.md (hunk 1 above)
Comment (hymm) [reply content/news/2023-03-04-bevy-0.10/index.md]: tried to add more sentences instead. Does it read better now?

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 9]
```diff
@@ -501,6 +501,39 @@ As this brings Bevy closer to full support of Android, there isn't a need anymor
...
+
+<div class="release-feature-authors">authors: @hymm, @james7132</div>
+
+![Trace with Pipelined Rendering](pipelined-rendering-trace.png)
+
+On multithreaded platforms, bevy will now run significantly faster by running simulation and
+rendering in parallel. The renderer was rearchitected in [bevy 0.6](https://bevyengine.org/news/bevy-0-6/#pipelined-rendering-extract-prepare-queue-render)
+to enable this, but the final step of actually running them in parallel was not done until now.
+There was a bit of tricky work to figure out. The render world has a system that has to run on
+the main thread, but task pool only had the ability to run on the world's thread. So when we send
```
Comment (Trouv) [inline content/news/2023-03-04-bevy-0.10/index.md:514]:
```suggestion
the main thread, but the task pool only had the ability to run on the world's thread. So, when we send
```

File: content/news/2023-03-04-bevy-0.10/index.md (hunk 1 above)
Comment (Trouv) [reply content/news/2023-03-04-bevy-0.10/index.md]: Yes it does, though I wonder if it's going into too much detail about the implementation now. Not 100% sure what level of detail is intended for this post

**Review by james7132 — APPROVED** [review]
LGTM sans @Trouv's comments.

**Review by cart — APPROVED** [review]
